diff --git a/drivers/media/dvb-frontends/gx1503.c b/drivers/media/dvb-frontends/gx1503.c
index 8107d63c2ce8..c01701afab2c 100644
--- a/drivers/media/dvb-frontends/gx1503.c
+++ b/drivers/media/dvb-frontends/gx1503.c
@@ -553,7 +553,7 @@ static int gx1503_probe(struct i2c_client *client)
 	return ret;
 }
 
-static void gx1503_remove(struct i2c_client *client)
+static int gx1503_remove(struct i2c_client *client)
 {
 	struct gx1503_dev*dev = i2c_get_clientdata(client);
 	
@@ -567,6 +567,7 @@ static void gx1503_remove(struct i2c_client *client)
 	dev->fe.ops.release = NULL;
 	dev->fe.demodulator_priv = NULL;
 	kfree(dev);
+	return 0;
 }
 
 static const struct i2c_device_id gx1503_id_table[] = {
diff --git a/drivers/media/dvb-frontends/m88rs6060.c b/drivers/media/dvb-frontends/m88rs6060.c
index 265632489eef..185bee7eeb29 100644
--- a/drivers/media/dvb-frontends/m88rs6060.c
+++ b/drivers/media/dvb-frontends/m88rs6060.c
@@ -3595,7 +3595,7 @@ static int m88rs6060_probe(struct i2c_client *client)
 	return ret;
 }
 
-static void m88rs6060_remove(struct i2c_client *client)
+static int m88rs6060_remove(struct i2c_client *client)
 {
 	struct m88rs6060_dev *dev = i2c_get_clientdata(client);
 	
@@ -3619,6 +3619,7 @@ static void m88rs6060_remove(struct i2c_client *client)
 	dev->fe.demodulator_priv = NULL;
 
 	kfree(dev);
+	return 0;
 }
 
 static const struct i2c_device_id m88rs6060_id_table[] = {
diff --git a/drivers/media/dvb-frontends/mtv23x.c b/drivers/media/dvb-frontends/mtv23x.c
index 939b2babd353..d560799bfa51 100644
--- a/drivers/media/dvb-frontends/mtv23x.c
+++ b/drivers/media/dvb-frontends/mtv23x.c
@@ -2470,12 +2470,13 @@ static int mtv23x_probe(struct i2c_client *client)
 	return ret;
 }
 
-static void mtv23x_remove(struct i2c_client *client)
+static int mtv23x_remove(struct i2c_client *client)
 {
 	struct mtv23x_dev*dev = i2c_get_clientdata(client);
 
 	regmap_exit(dev->regmap);
 	kfree(dev);
+	return 0;
 }
 
 static const struct i2c_device_id mtv23x_id_table[] ={
diff --git a/drivers/media/dvb-frontends/si2183.c b/drivers/media/dvb-frontends/si2183.c
index fd1e78d9f0d8..625e6eda6f0f 100644
--- a/drivers/media/dvb-frontends/si2183.c
+++ b/drivers/media/dvb-frontends/si2183.c
@@ -1795,7 +1795,7 @@ static int si2183_probe(struct i2c_client *client)
 	return ret;
 }
 
-static void si2183_remove(struct i2c_client *client)
+static int si2183_remove(struct i2c_client *client)
 {
 	struct si2183_dev *dev = i2c_get_clientdata(client);
 
@@ -1818,6 +1818,7 @@ static void si2183_remove(struct i2c_client *client)
 	dev->fe.demodulator_priv = NULL;
 
 	kfree(dev);
+	return 0;
 }
 
 static const struct i2c_device_id si2183_id_table[] = {
diff --git a/drivers/media/tuners/mxl603.c b/drivers/media/tuners/mxl603.c
index ae5a05aa28bc..3b29d3af0f11 100644
--- a/drivers/media/tuners/mxl603.c
+++ b/drivers/media/tuners/mxl603.c
@@ -555,11 +555,12 @@ static int mxl603_probe(struct i2c_client *client)
 	return ret;
 }
 
-static void mxl603_remove(struct i2c_client *client)
+static int mxl603_remove(struct i2c_client *client)
 {
 	struct mxl603_dev *dev = i2c_get_clientdata(client);
 
 	kfree(dev);
+	return 0;
 }
 
 static const struct i2c_device_id mxl603_id_table[] = {
